Wikipedia/assets/codemirror/resources/addon/search/search.js

```diff
@@ -30,6 +34,15 @@
         var match = query.exec(stream.string);
         if (match && match.index == stream.pos) {
           stream.pos += match[0].length || 1;
+
+          var fromCursor = cm.getCursor('from')
+          if (stream.lineOracle.line > fromCursor.line || (stream.lineOracle.line == fromCursor.line && stream.start >= fromCursor.ch)) {
```

Comment:
This is the main logic for determining if this match is the first one after the cursor and saving the index in a state property. They were just available values I saw to me in a breakpoint, not determined from any documentation...but it seems to work pretty well. It's worth noting that our find & replace implementation doesn't lean on setting the selection to highlight the match, so these are workarounds for that existing implementation. Seems like the way our native menus react are based on selection so I didn't want to hijack that to match the codemirror find/replace demo.